pkg/sql/scrub.go

```diff
@@ -543,21 +548,83 @@ func scrubPlanDistSQL(
 	return &physPlan, nil
 }
 
-// scrubRunDistSQL run a distSQLPhysicalPlan plan in distSQL. If
-// RowContainer is returned, the caller must close it.
+var _ rowResultWriter = &channelRowContainer{}
+
+// channelRowContainer implements the rowResultWriter interface. It is a
+// simple implementation of a rowResultWriter that streams and buffers
+// results using a buffered channel.
+type channelRowContainer struct {
+	results chan tree.Datums
+	status  error
+}
+
+// channelRowContainerBufferSize is the size of the buffer for the row
+// container.
+const channelRowContainerBufferSize = 1024
+
+func newChannelRowContainer() *channelRowContainer {
+	return &channelRowContainer{
+		results: make(chan tree.Datums, channelRowContainerBufferSize),
+	}
+}
+
+// AddRow implements the rowResultWriter interface.
+func (c *channelRowContainer) AddRow(ctx context.Context, row tree.Datums) error {
+	// Rows are copied because the Datums memory is re-used for the
+	// next result in a different routine then the consumer.
+	// FIXME(joey): Do we need to do any memory accounting here?
```

Comment:
Like this says, I wasn't sure if we need to do any memory accounting here. The flow is a bit weird here too, as there isn't a time when the memory is considered cleared (except maybe by a `planNode.Close` calling a new `channelRowContainer.Clear`).